document/core/appendix/changes.rst

```diff
@@ -7,6 +7,39 @@ Change History
 Since the original release 1.0 of the WebAssembly specification, a number of proposals for extensions have been integrated.
 The following sections provide an overview of what has changed.
 
+All present and future versions of WebAssembly are intended to be *backwards-compatible* with all previous versions.
+Concretely:
+
+1. All syntactically :ref:`well-formed <binary>` and :ref:`valid <valid>` modules remain well-formed and valid with an equivalent :ref:`module type <syntax-moduletype>` (or a subtype).
+
+  .. note::
+     This allows previously malformed or invalid modules to become legal,
+     e.g., by adding new features or by relaxing typing rules.
+
+     It also allows reclassifying previously malformed modules as well-formed but invalid,
+     or vice versa.
+
+     And it allows refining the typing of :ref:`imports <syntax-import>` and :ref:`exports <syntax-export>`,
+     such that previously unlinkable modules become linkable.
+
+2. All non-:ref:`trapping <trap>` :ref:`executions <exec>` of a valid program retain their behaviour with an equivalent set of possible :ref:`results <syntax-result>` (or a non-empty subset).
+
+  .. note::
+    This allows previously illegal programs to become executable.
+
+    It also allows program executions that previously trapped to execute successfully.
```

Comment:
I'm not sure about this point. Have we utilized this to extend semantics to date?

One reason to avoid that is that trapping is observable in an embedding context, and both Wasm programs and embedded environments can use trapping operations as security checks; i.e. a program won't continue past a divide by zero or and out-of-bounds memory access and continue executing in an unanticipated state.

Reply:
It seems useful to at least guarantee (if we can make such guarantees) that at least some traps will never be lifted, at the very least for the `unreachable` instruction.

Would it be feasible to include this sense of "cases where we previously documented that certain trapping cases might have legal behaviour in the future" in the text here, and limit the guarantee to that?

Reply:
Added a qualification along these lines.